### PR TITLE
[IOTDB-5284] Fix some code smells

### PR DESCRIPTION
**Pull request by BUAAserein**
### Fix some code smells in node-commons, please see in [JIRA5284](https://issues.apache.org/jira/browse/IOTDB-5284)

### CODE REVIEW

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java [hunk 1]
```diff
@@ -49,7 +50,7 @@ public static String readString(ByteBuffer buffer) {
   public static List<String> readStringList(ByteBuffer buffer) {
     int size = readInt(buffer);
     if (size < 0) {
-      return null;
+      return Collections.emptyList();
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java:53]: You need to check for upper-level dependencies on these nulls, such as changing == null to isEmpty when serializing

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java [hunk 2]
```diff
@@ -197,7 +198,7 @@ public static int write(Map<String, String> map, DataOutputStream stream) throws
   public static Map<String, String> readMap(ByteBuffer buffer) {
     int length = readInt(buffer);
     if (length == -1) {
-      return null;
+      return Collections.emptyMap();
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java:201]: same

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java [hunk 3]
```diff
@@ -243,7 +244,7 @@ public static int writeStringMapLists(Map<String, List<String>> map, ByteBuffer
   public static Map<String, List<String>> readStringMapLists(ByteBuffer buffer) {
     int length = readInt(buffer);
     if (length == -1) {
-      return null;
+      return Collections.emptyMap();
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java:247]: same

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java [hunk 4]
```diff
@@ -285,7 +286,7 @@ public static int writeIntMapLists(Map<Integer, List<Integer>> map, ByteBuffer b
   public static Map<Integer, List<Integer>> readIntMapLists(ByteBuffer buffer) {
     int length = readInt(buffer);
     if (length == -1) {
-      return null;
+      return Collections.emptyMap();
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java:289]: same

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java (hunk 1 above)
Comment (BUAAserein) [reply node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java]: Already removed

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java (hunk 2 above)
Comment (BUAAserein) [reply node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java]: Already removed

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java (hunk 4 above)
Comment (BUAAserein) [reply node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java]: Already removed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**